Sources/SpeziHealthKitBulkExport/BulkExportSessionProtocol.swift

```diff
@@ -0,0 +1,89 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import HealthKit
+import SpeziHealthKit
+
+
+/// State of a ``BulkExportSession``
+public enum BulkExportSessionState: Hashable, Sendable {
+    /// The session is currently paused.
+    ///
+    /// This is also the initial state for newly created but not yet started sessions.
+    case paused
+    /// The session is currently running.
+    case running
+    /// The session has completed its work, and has nothing else left to do.
+    case done
+}
+
+
+/// Information about the current progress of a ``BulkExportSession``.
+public struct BulkExportSessionProgress: Sendable {
```

Comment:
I am wondering if we should expose a nice mechanism to get a Foundation `Progress` type here or somewhere else? Would make it easy to interface with some SwiftUI views, e.g., https://developer.apple.com/documentation/swiftui/progressview/init(_:)-l5vj. 

Also wondering if we should expose the number of failed batches here as well? And/or an error attached to each failed batch? Could be included in the progress indicator (also counts failed batches to it will reach 100%).

Reply:
we're now (this is one of the changes in response to our discussion yesterday) exposing all individual pending/failed/completed batches as part of the `ExportSessionProtocol`, which also gives access to the number of samples (`session.failedBatches.count`). we also have `session.pendingBatches` and `session.completedBatches`.

regarding the progress: i was actually thinking about getting rid of the `BulkExportSessionProgress` type (or at least cut it down to only the `batchDescription`, since the other elements (`#completed` and `#total`) are now exposed on the session directly. we could just have a `Progress?` property on the session, which would be `nil` for paused sessions, and nonnil for runnig sessions

Reply:
Good idea to move the progress right in the session. I like them to just be exposed right there.

And the `Progress` could just be a convenient computed property exposed in addition based on all the other values so a user can easily plug it into a SwiftUI view.

Reply:
since it's a class it can't be a computed property. [`Progress`](https://developer.apple.com/documentation/foundation/progress) isn't Observable, but it is KVO compliant (it's that old...) so my guess is that the `ProgressView` uses that internally when initializing it with a `Progress` instance

Reply:
Oh, interesting, so we would need to manually update the stored instance? Might still be valuable to do that but you are right, not be best use of modern Swift.

Reply:
would we want the `Progress` to be relative the the total number of batches, or should it exclude failed batches?

ie, if we have 10 batches, 8 of which were processed successfully and 2 of which failed (and were scheduled for an automated retry next launch or earlier if explicitly requested): should the progress at the end be 80% or at 100%?

Reply:
So it would mean that it would be 80% complete in the example? I think that makes sense. Would probably also make sense when we add an automatic retry in the future.